src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+
+                            modifiedMembers.Add(modifiedConstructor);
+                            continue;
+                        }
+
+                    case OperatorDeclarationSyntax op:
+                        {
+                            // keep track of equality overloads so we can potentially remove them later
+                            var opKind = op.OperatorToken.Kind();
+                            if (opKind == SyntaxKind.EqualsEqualsToken)
+                            {
+                                equals = op;
+                            }
+                            else if (opKind == SyntaxKind.ExclamationEqualsToken)
+                            {
+                                notEquals = op;
+                            }
+
+                            break;
+                        }
+
+                    case MethodDeclarationSyntax method:
+                        {
+                            var methodSymbol = semanticModel.GetDeclaredSymbol(method, cancellationToken);
+                            if (methodSymbol is null)
+                            {
+                                modifiedMembers.Add(method);
+                                continue;
+                            }
+
+                            if (methodSymbol.Name == "Clone")
+                            {
+                                // delete any 'clone' method as it is reserved in records
+                                continue;
+                            }
+
+                            // TODO: Ensure the single param is a StringBuilder
+                            if (methodSymbol.Name == "PrintMembers" &&
+                                methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                                methodSymbol.Parameters.Length == 1)
+                            {
+                                if (type.TypeKind == TypeKind.Class && !type.IsSealed)
+                                {
+                                    // ensure virtual and protected modifiers
+                                    // TODO: change from virtual to override if inheriting from a record
+                                    modifiedMembers.Add(method.WithModifiers(SyntaxFactory.TokenList(
+                                        SyntaxFactory.Token(SyntaxKind.ProtectedKeyword),
+                                        SyntaxFactory.Token(SyntaxKind.VirtualKeyword))));
+                                }
+                                else
+                                {
+                                    // ensure private member
+                                    modifiedMembers.Add(method.WithModifiers(
+                                        SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.PrivateKeyword))));
+                                }
```

Comment:
i'm a bit confused as to what's going on here.  needs explanation.  why do we care if the user has a PrintMembers helper?  if anything, i would expect we would jsut remove their member as the record can already print itself properly.

Reply:
The PrintMembers method has an auto-generated implementation unless the user provides an implementation. It is used in the autogenerated ToString method (which the user can also write themselves). However, if the user defines one of these members, it must be declared `protected/private virtual/override` (depending on whether the record inherits from another record or can be inherited), or else there is an error.

It feels like an odd requirement to have in the first place, and the same restrictions are on the property `Type EqualityContract { get; }`

I'm a bit torn as to what to do here if we see a PrintMembers helper that does not follow these rules (and by extension EqualityContract). On one hand it's really easy to change the modifiers and removes a potentially confusing/annoying error for the user. On the other hand, I don't really see why the user would have a `public PrintMembers` method with this exact signature other than by random chance, and it might be equally as confusing to see that we changed it.

Happy to hear thoughts about this.

Reply:
> I don't really see why the user would have a public PrintMembers method with this exact signature other than by random chance, and it might be equally as confusing to see that we changed it.

Agreed, it seems relatively unlikely that the user would have a member like this. My biggest concern is if the member is public, then it might be because the user is using it somewhere else to ask a type to print itself, and by making this non-public we're breaking that. Yes the user might have to rename the method (I agree the compiler limitation here is odd), but does making this private make it harder for the user to also rename the method and update uses?

Reply:
I think this makes sense. If the user actually wrote this method (either by chance or some "record generator" which would auto-generate the method), they likely would intentionally make it public. Furthermore, I think there's a variety of ways to handle the error (making protected, renaming, or deleting altogether). For now, I'll remove the handling on this method and the user can handle the error.